etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveAccessionNumber.java

```diff
@@ -0,0 +1,59 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Resource;
+
+public class RemoveAccessionNumber implements CustomFhirTransformation {
+    public static final String CODE_NAME = "code";
+    public static final String CODING_SYSTEM_NAME = "codingSystemExtension";
+    public static final String CODING_NAME = "codingExtension";
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args) {
+        var bundle = (Bundle) resource.getUnderlyingResource();
+        Set<Resource> resourcesToRemove = new HashSet<>();
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            Resource resourceEntry = entry.getResource();
+
+            if (resourceEntry instanceof Observation observation) {
+                processObservation(observation, resourcesToRemove, args);
+            }
+        }
+
+        bundle.getEntry().removeIf(entry -> resourcesToRemove.contains(entry.getResource()));
+    }
+
+    private void processObservation(
```

Comment:
Instead of `processObservation` it would be helpful to be more specific about that it does. Something like `removeRelevantObservations`? `Relevant` could be replaced by something more specific though

Reply:
Yes, we have some refactoring to do still.